src/event-bridge/event-bus.ts

```diff
@@ -345,6 +356,100 @@ abstract class EventBusBase<in Evnt extends Event, OutEvnt extends Evnt = Evnt>
           });
         });
       },
+      apiGWVtl: {
+        renderRequest: (call, context): string => {
+          const args = call.args
+            .map((arg) => arg.expr)
+            .filter((arg): arg is Expr => !!arg)
+            .flatMap((arg) => (isArrayLiteralExpr(arg) ? arg.items : arg));
+
+          context.set(
+            "$context.requestOverride.header.X-Amz-Target",
+            '"AWSEvents.PutEvents"'
+          );
+          context.set(
+            "$context.requestOverride.header.Content-Type",
+            '"application/x-amz-json-1.1"'
+          );
+
+          const argObjects = args.map((arg) => {
+            if (!isObjectLiteralExpr(arg)) {
+              throw new SynthError(
+                ErrorCodes.Expected_an_object_literal,
+                "API Gateway Integration with EventBus.putEvents expects object literals with no computed properties"
+              );
+            }
+
+            const objectProps = arg.properties.map((prop) => {
+              if (
+                !isPropAssignExpr(prop) ||
+                !(isIdentifier(prop.name) || isStringLiteralExpr(prop.name))
```

Comment:
nit: are the parens necessary?

Reply:
Would it be easier to check for what's allowed instead of the `!`? I am confused.

Reply:
what about numeric properties?

Reply:
![image](https://user-images.githubusercontent.com/289213/181172065-88e18ebd-feaf-4c7e-b367-e3a2a308c8d9.png)

Looks like that is our miss

`export type PropertyName = Identifier | StringLiteral | NumericLiteral | ComputedPropertyName | PrivateIdentifier`